daly-bms-uart.cpp

```diff
@@ -39,63 +40,277 @@ bool Daly_BMS_UART::Init()
+    alarm.levelOneStateOfChargeTooHigh = bitRead(this->my_rxBuffer[6], 4);
+    alarm.levelTwoStateOfChargeTooHigh = bitRead(this->my_rxBuffer[6], 5);
+    alarm.levelOneStateOfChargeTooLow = bitRead(this->my_rxBuffer[6], 6);
+    alarm.levelTwoStateOfChargeTooLow = bitRead(this->my_rxBuffer[6], 7);
+
+    /* 0x03 */
+    alarm.levelOneCellVoltageDifferenceTooHigh = bitRead(this->my_rxBuffer[7], 0);
+    alarm.levelTwoCellVoltageDifferenceTooHigh = bitRead(this->my_rxBuffer[7], 1);
+    alarm.levelOneTempSensorDifferenceTooHigh = bitRead(this->my_rxBuffer[7], 2);
+    alarm.levelTwoTempSensorDifferenceTooHigh = bitRead(this->my_rxBuffer[7], 3);
+
+    /* 0x04 */
+    alarm.chargeFETTemperatureTooHigh = bitRead(this->my_rxBuffer[8], 0);
+    alarm.dischargeFETTemperatureTooHigh = bitRead(this->my_rxBuffer[8], 1);
+    alarm.failureOfChargeFETTemperatureSensor = bitRead(this->my_rxBuffer[8], 2);
+    alarm.failureOfDischargeFETTemperatureSensor = bitRead(this->my_rxBuffer[8], 3);
+    alarm.failureOfChargeFETAdhesion = bitRead(this->my_rxBuffer[8], 4);
+    alarm.failureOfDischargeFETAdhesion = bitRead(this->my_rxBuffer[8], 5);
+    alarm.failureOfChargeFETTBreaker = bitRead(this->my_rxBuffer[8], 6);
+    alarm.failureOfDischargeFETBreaker = bitRead(this->my_rxBuffer[8], 7);
+
+    /* 0x05 */
+    alarm.failureOfAFEAcquisitionModule = bitRead(this->my_rxBuffer[9], 0);
+    alarm.failureOfVoltageSensorModule = bitRead(this->my_rxBuffer[9], 1);
+    alarm.failureOfTemperatureSensorModule = bitRead(this->my_rxBuffer[9], 2);
+    alarm.failureOfEEPROMStorageModule = bitRead(this->my_rxBuffer[9], 3);
+    alarm.failureOfRealtimeClockModule = bitRead(this->my_rxBuffer[9], 4);
+    alarm.failureOfPrechargeModule = bitRead(this->my_rxBuffer[9], 5);
+    alarm.failureOfVehicleCommunicationModule = bitRead(this->my_rxBuffer[9], 6);
+    alarm.failureOfIntranetCommunicationModule = bitRead(this->my_rxBuffer[9], 7);
+
+    /* 0x06 */
+    alarm.failureOfCurrentSensorModule = bitRead(this->my_rxBuffer[10], 0);
+    alarm.failureOfMainVoltageSensorModule = bitRead(this->my_rxBuffer[10], 1);
+    alarm.failureOfShortCircuitProtection = bitRead(this->my_rxBuffer[10], 2);
+    alarm.failureOfLowVoltageNoCharging = bitRead(this->my_rxBuffer[10], 3);
+
+    return true;
+}
+
+bool Daly_BMS_UART::setMOSGate(bool sw) //0xD9 0x80 First Byte 0x01=ON 0x00=OFF
+{
+
+    /**
+     * it works
+     * but it must give a way to switch the charge and discharge gate seperate
+     */
+    if (sw == true)
+    {
+        #ifdef DALY_BMS_DEBUG
+        DEBUG_SERIAL.println("try switching discharge on");
+        #endif
+        this->my_txBuffer[4] = 0x01;
+        this->sendCommand(COMMAND::DISCHRG_FET);
+        this->my_txBuffer[4] = 0x00;
+    }
+
+    if (sw == false)
+    {
+        #ifdef DALY_BMS_DEBUG
+        DEBUG_SERIAL.println("try switching discharge off");
+        #endif
+        this->sendCommand(COMMAND::DISCHRG_FET);
+    }
+    if (!receiveBytes())
```

Comment:
Do you know if we expect a specific response to this command?

Reply:
no, it is visibil with the next bms cycle in the status fromt he fet
![grafik](https://user-images.githubusercontent.com/44615614/144823572-934c45c5-8d4c-45f2-a746-bb7bb4e86f9f.png)
![grafik](https://user-images.githubusercontent.com/44615614/144823694-4c343550-4b06-45b5-9815-775fefb4a573.png)
not sure how to work with this response